src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -24,133 +24,59 @@
-        $data['defaults'] = $data['defaults'] ?? $defaults;
-        $data['host'] = $data['host'] ?? $host;
-        $data['methods'] = $data['methods'] ?? $methods;
-        $data['schemes'] = $data['schemes'] ?? $schemes;
-        $data['condition'] = $data['condition'] ?? $condition;
-        $data['priority'] = $data['priority'] ?? $priority;
-        $data['locale'] = $data['locale'] ?? $locale;
-        $data['format'] = $data['format'] ?? $format;
-        $data['utf8'] = $data['utf8'] ?? $utf8;
-        $data['stateless'] = $data['stateless'] ?? $stateless;
-        $data['env'] = $data['env'] ?? $env;
-
-        $data = array_filter($data, static function ($value): bool {
-            return null !== $value;
-        });
-
-        if (isset($data['localized_paths'])) {
-            throw new \BadMethodCallException(sprintf('Unknown property "localized_paths" on annotation "%s".', static::class));
-        }
-
-        if (isset($data['value'])) {
-            $data[\is_array($data['value']) ? 'localized_paths' : 'path'] = $data['value'];
-            unset($data['value']);
-        }
-
-        if (isset($data['path']) && \is_array($data['path'])) {
-            $data['localized_paths'] = $data['path'];
-            unset($data['path']);
-        }
-
-        if (isset($data['locale'])) {
-            $data['defaults']['_locale'] = $data['locale'];
-            unset($data['locale']);
+        if (\is_array($path)) {
+            $this->localizedPaths = $path;
+        } else {
+            $this->path = $path;
         }
+        $this->setMethods($methods);
+        $this->setSchemes($schemes);
 
-        if (isset($data['format'])) {
-            $data['defaults']['_format'] = $data['format'];
-            unset($data['format']);
+        if (null !== $locale) {
+            $this->defaults['_locale'] = $locale;
         }
 
-        if (isset($data['utf8'])) {
-            $data['options']['utf8'] = filter_var($data['utf8'], \FILTER_VALIDATE_BOOLEAN) ?: false;
-            unset($data['utf8']);
+        if (null !== $format) {
+            $this->defaults['_format'] = $format;
         }
 
-        if (isset($data['stateless'])) {
-            $data['defaults']['_stateless'] = filter_var($data['stateless'], \FILTER_VALIDATE_BOOLEAN) ?: false;
-            unset($data['stateless']);
+        if (null !== $utf8) {
+            $this->options['utf8'] = $utf8;
         }
 
-        foreach ($data as $key => $value) {
-            $method = 'set'.str_replace('_', '', $key);
-            if (!method_exists($this, $method)) {
-                throw new \BadMethodCallException(sprintf('Unknown property "%s" on annotation "%s".', $key, static::class));
-            }
-            $this->$method($value);
+        if (null !== $stateless) {
+            $this->defaults['_stateless'] = $stateless;
         }
     }
 
-    public function setPath($path)
+    public function setPath(string $path)
```

Comment:
I can't remember, why didn't we add these in 5.*

Reply:
Me neither, but I felt we should do this now.